GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitCommandsTests.cs

```diff
@@ -415,110 +416,52 @@ public void CaseOnlyRenameFileAndChangeBranches()
             this.FileShouldHaveCaseMatchingName(newFileName);
         }
 
+        // Mac and Windows only because Linux uses a separate repo mount device
         [TestCase]
-        public void MoveFileFromOutsideRepoToInsideRepoAndAdd()
+        [Category(Categories.RepositoryMountsSameFileSystem)]
+        public void MoveFileFromOutsideRepoToInsideRepoOnSameFileSystemAndAdd()
         {
-            string testFileContents = "0123456789";
-            string filename = "MoveFileFromOutsideRepoToInsideRepo.cs";
-
-            // Create the test files in this.Enlistment.EnlistmentRoot as it's outside of src and the control
-            // repo and is cleaned up when the functional tests run
-            string oldFilePath = Path.Combine(this.Enlistment.EnlistmentRoot, filename);
-            string controlFilePath = Path.Combine(this.ControlGitRepo.RootPath, filename);
-            string gvfsFilePath = Path.Combine(this.Enlistment.RepoRoot, filename);
-
-            string newBranchName = "tests/functional/MoveFileFromOutsideRepoToInsideRepoAndAdd";
-            this.ValidateGitCommand("checkout -b " + newBranchName);
-
-            // Move file to control repo
-            this.FileSystem.WriteAllText(oldFilePath, testFileContents);
-            this.FileSystem.MoveFile(oldFilePath, controlFilePath);
-            oldFilePath.ShouldNotExistOnDisk(this.FileSystem);
-            controlFilePath.ShouldBeAFile(this.FileSystem).WithContents(testFileContents);
-
-            // Move file to GVFS repo
-            this.FileSystem.WriteAllText(oldFilePath, testFileContents);
-            this.FileSystem.MoveFile(oldFilePath, gvfsFilePath);
-            oldFilePath.ShouldNotExistOnDisk(this.FileSystem);
-            gvfsFilePath.ShouldBeAFile(this.FileSystem).WithContents(testFileContents);
-
-            this.ValidateGitCommand("status");
-            this.ValidateGitCommand("add .");
-            this.RunGitCommand("commit -m \"Change for MoveFileFromOutsideRepoToInsideRepoAndAdd\"");
-            this.ValidateGitCommand("checkout " + this.ControlGitRepo.Commitish);
+            this.MoveFileFromOutsideRepoToInsideRepoAndAdd(true);
         }
 
+        // Linux only because Linux uses a separate repo mount device
         [TestCase]
-        public void MoveFolderFromOutsideRepoToInsideRepoAndAdd()
+        [Category(Categories.RepositoryMountsDifferentFileSystem)]
+        public void MoveFileFromOutsideRepoToInsideRepoOnDifferentFileSystemAndAdd()

```

Comment:
Since these kinds of activities are effectively impermissible on Linux, I think the same logic above holds and so I've just marked these as `RepositoryMountsSameFileSystem` and left it at that.